packages/fuels-core/src/tokenizer.rs

```diff
@@ -55,7 +53,14 @@ impl Tokenizer {
             ParamType::Struct {
                 fields: struct_params,
                 ..
-            } => Ok(Self::tokenize_struct(trimmed_value, struct_params)?),
+            } => Ok(Self::tokenize_struct(
+                trimmed_value,
+                &struct_params
+                    .iter()
+                    .map(|(_, param_type)| param_type)
+                    .cloned()
+                    .collect::<Vec<_>>(),
```

Comment:
The fact that this `iter().map().clone().collect()` construction appears so many times makes me think there should be an associated function or something

Reply:
I do not know of another way. We cannot use `into_iter` because it would move the values, so we have to clone it at some point..

Reply:
I was not talking about the construction itself (I tried to improve it via `unzip` but couldn't haha), but the fact that maybe we should have a function associated to the `struct_params` that does this -> it's basically an unzip. If I'm not clear I can try to come up with a code example.

Reply:
Oh ok i  got you now. I thought you were looking for an iterator function that would do these steps :smile: I will add it somewhere.